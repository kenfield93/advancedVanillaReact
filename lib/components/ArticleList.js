/**
 * Created by kyle on 9/28/17.
 */
import React, {PropTypes} from 'react';
import Article from './Article';

//Refactor to presentatoin/dumb component
//also destruct authors, articles from props
class ArticleList extends React.Component{
    constructor(props){
        super(props);

    }

    render(){
        const KEY = 0, VALUE = 1;
        return(
            <div>
                { Array.from(this.props.articles).map( article => {
                    article = article[VALUE];
                    return (
                        <Article key={article.id}
                                 article={article}
                                 actions={this.props.authorActions}>
                        </Article>
                    );}
                )}
            </div>
        );
    }
}

ArticleList.propTypes = {
    articles: PropTypes.object.isRequired,
    authorActions: PropTypes.object.isRequired
};
export default ArticleList;