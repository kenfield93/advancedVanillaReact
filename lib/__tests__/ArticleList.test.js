/**
 * Created by kyle on 9/28/17.
 */
import React from 'react';
import ArticleList from '../components/ArticleList';

import render from 'react-test-renderer';

describe('Testing ArticleList', ()=>{
    const articles = new Map();
    articles.set('a', {id: 'a'});
    articles.set('b', {id: 5});

    const testProps = {
       articles,
       authorActions: {
           getAuthor: (authorId) => {return {
               firstName: 'George',
               lastName: 'Costanza',
               website: 'google.com'
           }}
       }
   };
   it('renders correctly', () =>{
        const tree = render.create(
            <ArticleList
                {...testProps}
            />
        ).toJSON();

        expect(tree.children.length).toEqual(testProps.articles.size);
        expect(tree).toMatchSnapshot();
   });
});