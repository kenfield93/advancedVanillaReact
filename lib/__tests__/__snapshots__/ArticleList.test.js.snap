// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ArticleList renders correctly 1`] = `
<div>
  <div
    style={
      Object {
        "borderBottomColor": "#aaaeef",
        "borderBottomStyle": "solid",
        "borderBottomWidth": 1,
        "marginBottom": 10,
        "paddingBottom": 10,
      }
    }
  >
    <h2
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    />
    <p
      style={
        Object {
          "color": "#888",
          "fontSize": "0.85em",
        }
      }
    >
      Invalid Date
       
    </p>
    <div
      style={
        Object {
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    >
      Written by: 
      <a
        href="google.com"
      >
        <i>
          George
           
          Costanza
        </i>
         
      </a>
    </div>
    <p
      style={
        Object {
          "paddingLeft": 20,
        }
      }
    >
       
    </p>
  </div>
  <div
    style={
      Object {
        "borderBottomColor": "#aaaeef",
        "borderBottomStyle": "solid",
        "borderBottomWidth": 1,
        "marginBottom": 10,
        "paddingBottom": 10,
      }
    }
  >
    <h2
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    />
    <p
      style={
        Object {
          "color": "#888",
          "fontSize": "0.85em",
        }
      }
    >
      Invalid Date
       
    </p>
    <div
      style={
        Object {
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    >
      Written by: 
      <a
        href="google.com"
      >
        <i>
          George
           
          Costanza
        </i>
         
      </a>
    </div>
    <p
      style={
        Object {
          "paddingLeft": 20,
        }
      }
    >
       
    </p>
  </div>
</div>
`;
